### DISCUSSION TITLE
Discussions on the Restlet Framework project!

### DISCUSSION THREAD

**Discussion by jlouvel** [restlet/restlet-framework-java#1382] [General]
## 👋 Welcome!
  While the official forum remains [this Mailing List](https://groups.google.com/a/restlet.org/g/framework-discuss), we'll be experimenting with GitHub Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Tembrel** [comment]
Tim Peierls here, long-time user of Restlet. Over the years I contributed various ideas, advice, and bits of code, including what became the Guice-Restlet extension (which should have been called the `javax.inject`-Restlet extension, but that's another story). [My blog](https://tembrel.blogspot.com/search/label/Restlet) is woefully out of date, but it does describe some of my contributions.

Seat Yourself, the tiny company that I founded with a friend nearly two decades ago, provides an online ticketing service for school and community performing arts groups. The service is fundamentally Restlet-based, consisting of a single Component running co-operating Applications. We're using Restlet Framework 2.4.3 running under Tomcat 8.5 / Java 8 in an AWS Elastic Beanstalk cluster.

**Reply by jlouvel** [reply]
Thanks Time for sharing your Restlet story and your continued contribution to the project!

For the **org.restlet.ext.guice** extension, would it make sense to rename it to **org.restlet.ext.inject** in the upcoming 2.5 version?

**Reply by Tembrel** [reply]
Ha, yes, only about 10 years late!

**Comment by kerbymart** [comment]
Hi @Tembrel what do you think of a new Dagger2 extension: https://github.com/restlet/restlet-framework-java/issues/1411

**Reply by Tembrel** [reply]
Rather than create a new extension for every `javax.inject` implementation out there, I think it would be better to add support within the `org.restlet.ext.inject` extension. I regret that this extension was originally named `org.restlet.ext.guice`, because the intent was to provide examples of how to get resources injected under **any** `javax.inject`-compliant system. I focused on Guice because I was using (and still use) Guice.

I'm not at all sure this stuff rightly belongs in an extension in the first place. Apart from `ServerResource` instances, the creation of which is handled by `Finder` machinery, the user is responsible for constructing all other Restlet classes, which makes Restlet a fine candidate for dependency injection. That's what [this section of the extension docs](https://restlet.talend.com/documentation/user-guide/2.4/extensions/guice#:~:text=Injecting%20other%20Restlet%20types) is about. 

The goal, as I see it, should be to allow the use of `javax.inject` annotations with any reference to a particular dependency injection framework confined to a small piece of configuration code.

**Comment by kerbymart** [comment]
@jlouvel Hello, I think creating a repository in the Restlet Github dedicated to "app starts" would be beneficial. This would allow developers who cannot contribute to the framework itself (the "users") to contribute by creating quick-start apps for JEE, GAE, and other modules. For example, a good start would be to have a "restlet/restlet-examples" repository.

**Reply by jlouvel** [reply]
Hi @kerbymart Thanks for the initiative, We do have an existing org.restlet.example module inside the main repository that has a similar purpose that that would benefit from a clean-up and probably becoming a separate repository as you suggest. Developing and incorporating additional targeted examples to get users started sounds beneficial.

**Comment by xSAVIKx** [comment]
Hey @jlouvel, that's great to see something happening in the framework! I hope there'd be some opportunity for the community to be more involved into the development and maybe faster release cycles.

I have a fork of my own that I had to create to fix some things and update dependencies. One thing I guess may be of interest is migration to maven-based deployments compared to ant-based ones. I'd be glad to donate that part to the main repo while I already have both maven central and github packages deployments ready and implemented [here](https://github.com/xSAVIKx/restlet-framework-java).

Let me know if that's smth worth back-merging.

**Reply by xSAVIKx** [reply]
I've also migrated builds to GH actions while Travis seems a bit less stable/easy to use approach nowadays

**Reply by jlouvel** [reply]
Hi @xSAVIKx thanks for the feedback and opportunity to contribute your code back.

We have started to work on the Mavenified build on [2.5 branch](https://github.com/restlet/restlet-framework-java/tree/2.5) but are far from done, so your contribution would be very timely.

Have you had a JCA signed already?
If not that could be a great step to take so we don't have to redevelop those wheels.

I'll send a formal message to the community to share our plans and seek feedback and contributions soon. Stay tuned

**Reply by xSAVIKx** [reply]
Yeah, I've glanced that you already have done a big chunk of it 👍

I'm not sure I understood what you meant by `Have you had a JCA signed already?`. What does that mean? 
Are you asking if the deployed jars are signed when being deployed to the central? If so - yes, the signing is part of the maven build.

What is the best way to contribute? Just opening a PR into the 2.5 would be ok?

**Reply by jlouvel** [reply]
Here is the page that explains what a JCA is and how to download and return it. Thanks and let us know if you have any question!
https://github.com/restlet/restlet-framework-java/wiki/Contribute

In term of the code contributing, providing a PR on the branch would be ideal. @thboileau is currently working on restoring a few additional modules from 2.4, but we could already complete the Maven build as you have already explored in your fork.

**Reply by xSAVIKx** [reply]
Hey @jlouvel, I created this [PR](https://github.com/restlet/restlet-framework-java/pull/1414) with the changes I think are relevant. You guys did a great job moving most of the things already 👍 

Please let me know if you're interested in reviewing/merging that PR and how can I share the signed JCA with you guys.
